pkg/intermediate/aggregate.go

```diff
@@ -118,74 +130,150 @@ func (a *AggregationProcess) DeleteFlowKeyFromMap(flowKey FlowKey) {
 	delete(a.flowKeyRecordMap, flowKey)
 }
 
-// correlateRecords fills records info by correlating incoming and current records
-func (a *AggregationProcess) correlateRecords(flowKey FlowKey, record entities.Record) {
+// aggregateRecord either adds the record to flowKeyMap or update the record in
+// flowKeyMap by doing correlation or updating the stats.
+func (a *AggregationProcess) aggregateRecord(flowKey FlowKey, record entities.Record) {
 	a.mutex.Lock()
 	defer a.mutex.Unlock()
-	existingRecords := a.flowKeyRecordMap[flowKey]
-	// only fill the information for record from source node
-	if isRecordFromSrc(record) {
-		var isFilled bool
-		for _, existingRec := range existingRecords {
-			for _, field := range a.correlateFields {
-				if ieWithValue, exist := record.GetInfoElementWithValue(field); exist {
-					existingIeWithValue, _ := existingRec.GetInfoElementWithValue(field)
-					ieWithValue.Value = existingIeWithValue.Value
-					isFilled = true
-				}
-			}
-			if isFilled {
-				break
+
+	aggregationRecord, exist := a.flowKeyRecordMap[flowKey]
+	if exist {
+		if !isRecordIntraNode(record) {
+			// Do correlation of records if record belongs to inter-node flow and
+			// records from source and destination node are not received.
+			if !aggregationRecord.ReadyToSend && !areRecordsFromSameNode(record, aggregationRecord.Record) {
+				a.correlateRecords(record, aggregationRecord.Record)
+				aggregationRecord.ReadyToSend = true
+			} else {
+				// If the record from the node is already present, update the stats
+				// and timestamps.
```

Comment:
We are not discarding the record as the previous record from the same node (either source/destination) will have the same flow key fields and metadata fields except for new stats and timestamps.

Reply:
Got it. we will only update existing record based on upcoming record. what if the existing record has not been exported and the upcoming record comes? will flow aggregator export one record instead of two?

Reply:
Yes, exporter in flow aggregator will export one record instead of two. The most updated version of it.